.travis.yml

```diff
@@ -4,7 +4,7 @@ dist: bionic
 language: python
 env:
   global:
-    - LSR_ANSIBLES='ansible==2.7.* ansible==2.8.* ansible==2.9.*'
```

Comment:
Not testing Ansible 2.7 should be justified by it being EOL. It could also be in its own PR/commit. Not testing it because molecule does not support it does not seem to be a valid justfication.

Reply:
`LSR_ANSIBLE` is *only* used for molecule testing

Reply:
It is required in order to move to molecule v3 - we have to drop support for `LSR_ANSIBLES='ansible==2.7.*'`

Reply:
So to me it logically must be part of the commit to upgrade to molecule v3

Reply:
split this into its own commit

Reply:
If we need to support Ansible 2.7 with the system roles, this would be a blocker for adopting molecule V3, I guess. Therefore an explanation why Ansible 2.7 testing can be dropped would be nice in the commit message.

Another note:
Also to keep the repo always in a consistent state, it would be nicer to first apply the commits that prepare molecule v3 support and once this is done, switch to molecule v3. This will keep git bisect useful.

Reply:
> If we need to support Ansible 2.7 with the system roles, this would be a blocker for adopting molecule V3, I guess. 

Why?  This is only for molecule.

I suppose if it is necessary (as determined by whom?) that we must have molecule support for Ansible 2.7, we can evaluate supporting both molecule v2 *and* molecule v3 in tox/travis.

The way I see it, it isn't necessary that the molecule testing in CI covers all of the Ansible versions we support.  It is another testing tool that gives us some coverage, and any coverage that it doesn't provide, it is up to us to fill in the gaps in molecule coverage via some other method.

>Therefore an explanation why Ansible 2.7 testing can be dropped would be nice in the commit message.

How about what I said above ^^^?

> 
> Another note:
> Also to keep the repo always in a consistent state, it would be nicer to first apply the commits that prepare molecule v3 support and once this is done, switch to molecule v3. This will keep git bisect useful.

I thought I did that.  Can you explain what you mean by "the commits that prepare molecule v3 support", and how the current commits I have can leave the repo in an inconsistent state?  Do you mean "remove redundant flake8, yamllint checks from molecule" which can be applied first?  Can you explain why the current order of commits will cause problems with `git bisect`?

Reply:
> > If we need to support Ansible 2.7 with the system roles, this would be a blocker for adopting molecule V3, I guess.
> 
> Why? This is only for molecule.
> 
> I suppose if it is necessary (as determined by whom?) that we must have molecule support for Ansible 2.7, we can evaluate supporting both molecule v2 _and_ molecule v3 in tox/travis.
> 
> The way I see it, it isn't necessary that the molecule testing in CI covers all of the Ansible versions we support. It is another testing tool that gives us some coverage, and any coverage that it doesn't provide, it is up to us to fill in the gaps in molecule coverage via some other method.
> 
> > Therefore an explanation why Ansible 2.7 testing can be dropped would be nice in the commit message.
> 
> How about what I said above ^^^?

IMHO the explanation should answer the question "why is it ok not to test Ansible 2.7 with molecule". If this test is optional and already covered by other CI parts, it would be a valid explanation. If we do not need to test Ansible 2.7 at all, because it is EOL, that would be another one. Maybe some documentation why we do molecule at all is missing to ensure that we all know why it is there. To me it appeared to be required not optional and adding some extra checks that we want that are not present elsewhere.

> > Another note:
> > Also to keep the repo always in a consistent state, it would be nicer to first apply the commits that prepare molecule v3 support and once this is done, switch to molecule v3. This will keep git bisect useful.
> 
> I thought I did that. Can you explain what you mean by "the commits that prepare molecule v3 support", and how the current commits I have can leave the repo in an inconsistent state? Do you mean "remove redundant flake8, yamllint checks from molecule" which can be applied first? Can you explain why the current order of commits will cause problems with `git bisect`?

Once the commit "support molecule v3" is applied, the result is broken because molecule v3 support also requires the removal of Ansible 2.7. So if git-bisect would select that commit, tests using molecule would always fail even if there is no change in the role that might have broken something. Therefore it would be nicer to first apply a commit to prepare for molecule v3 support (removing Ansible 2.7) and then do the actual change. Btw. I just realized that the commit is not about "supporting molecule v3" but "switching to molecule v3" since AFAIU, v2 will not be supported after that change.

Reply:
> IMHO the explanation should answer the question "why is it ok not to test Ansible 2.7 with molecule".

Ok, I'll see what I can come up with.

> If we do not need to test Ansible 2.7 at all, because it is EOL, that would be another one.

I think this is the case. @pcahyna ?

The other option is that we support both molecule v2 and v3 until such time as we can make a definitive determination that we will no longer support Ansible 2.7.

> Once the commit "support molecule v3" is applied, the result is broken because molecule v3 support also requires the removal of Ansible 2.7.

What about if I put the change to remove Ansible 2.7 from LSR_ANSIBLES in the same commit as "support molecule v3" (which should be retitled to "switch from molecule v2 to v3")?